http-server/src/main/java/de/unistuttgart/iaas/amyassist/amy/httpserver/Server.java

```diff
@@ -45,31 +47,76 @@
 /**
  * A class to create a http server
  * 
- * @author Christian Bräuner, Leon Kiefer
+ * @author Christian Bräuner, Leon Kiefer, Tim Neumann
  */
 @Service
-public class Server{
-	private final Logger logger = LoggerFactory.getLogger(Server.class);
+public class Server {
+	/** The name of the config used by this class */
+	public static final String CONFIG_NAME = "server.config";
+	/** The name of the property, which specifies the port */
+	public static final String PROPERTY_PORT = "port";
+	/** The name of the property, which specifies the root path of the server */
+	public static final String PROPERTY_ROOT_PATH = "root_path";
+	/** The name of the property, which specifies whether the server should bind to local host only. */
+	public static final String PROPERTY_LOCALHOST = "local_host_only";
 
+	/**
+	 * The ip used, when the server should only be accessible from localhost
+	 */
+	public static final String IP_LOCAL = "127.0.0.1";
+	/**
+	 * The ip used, when the server should be accessible from anywhere
+	 */
+	public static final String IP_GLOBAL = "0.0.0.0";
+
+	@Reference
+	private Logger logger;
+
+	/**
+	 * The dependency injection instance, needed for binding the di to the server
+	 */
 	@Reference
-	private ServiceLocator di;
+	ServiceLocator di;
+
 	private Set<Class<?>> restResources = new HashSet<>();
 	private HttpServer httpServer;
 
+	@Reference
+	private ConfigurationLoader configurationLoader;
+
 	/**
-	 * the URI of the server
+	 * @return the URI of the server
 	 */
-	public static final URI BASE_URI = URI.create("http://localhost:8080/rest");
+	private URI baseURI() {
+		Properties conf = this.configurationLoader.load(CONFIG_NAME);
+		int port = Integer.parseInt(conf.getProperty(PROPERTY_PORT, "8080"));
+		String root = conf.getProperty(PROPERTY_ROOT_PATH);
+		String local = conf.getProperty(PROPERTY_LOCALHOST);
```

Comment:
convert local here to the boolean

Reply:
That is done a few lines later, so that a null check can occur before.